src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -401,6 +416,19 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		return id;
 	}
 
+	private async doRunAllCells(codeCells: ICellModel[], editor: MainThreadNotebookEditor): Promise<boolean> {
+		try {
+			if (codeCells && codeCells.length > 0) {
+				for (let i = 0; i < codeCells.length; i++) {
+					await editor.runCell(codeCells[i]);
+				}
+			}
+		} catch {
+			return Promise.resolve(false);
```

Comment:
Just chekcing, Isn't it Promise.reject in catch block which is what you have done in Action?